### PR TITLE
fix: inder Veer ji bug

### PR DESCRIPTION
**Pull request by amitojsingh**
fixed from https://ryazio.notion.site/Shabadavli-Code-Review-Phase-1-b5de2e49f7ed4a79bc4e538a3dda5c08
1. Remove all magic numbers
2. Remove redundant null calls
3. remove consoles
4. Fixed Deep scan issues
5. Added try catch

### CODE REVIEW

File: src/pages/profile/index.tsx [hunk 1]
```diff
@@ -212,14 +219,17 @@ export default function Profile() {
...
     }
+
     if (user?.uid) {
       setIsLoading(false);
-      setName(user.displayName);
-      setUsername(user.username);
-      setVerifiable(!user.emailVerified);
+      setName(user?.displayName ?? '');
+      setUsername(user?.username ?? '');
+      setVerifiable(!(user?.emailVerified ?? true));
```
Comment (MySmilingTurban) [inline src/pages/profile/index.tsx:232]: should we change these 
`user?.displayName ?? ''`
to
`user?.displayName || ''`

**Review by MySmilingTurban — CHANGES_REQUESTED** [review]

File: src/components/signin/LogOut.tsx [hunk 2]
```diff
@@ -18,14 +18,14 @@ export default function LogOut() {
...
-        );
+      } catch (error) {
+        if (error instanceof Error) {
+          setErrorMessage(error.message);
+          showToastMessage(error.message, toast.POSITION.BOTTOM_RIGHT, true, true);
+        } else {
+          setErrorMessage('An unknown error occurred');
+          showToastMessage('An unknown error occurred', toast.POSITION.BOTTOM_RIGHT, true, true);
+        }
       }
```
[the remark this answers is not in the document]
Comment (amitojsingh) [reply src/components/signin/LogOut.tsx]: fixed

File: src/components/signin/SignIn.tsx [hunk 3]
```diff
@@ -26,12 +26,14 @@ export default function SignIn() {
 
   const signToggle = (e: FormEvent) => {
     e.preventDefault();
-    const switchElement = document.getElementsByClassName('switch')[0];
 
-    if (switchElement?.classList.contains('left-[4px]')) {
+    const switchElement = document.getElementsByClassName('switch')[0] as HTMLElement | null;
+
+    if (switchElement !== null) {
+
```
[the remark this answers is not in the document]
Comment (amitojsingh) [reply src/components/signin/SignIn.tsx]: fixed

File: src/components/signin/SignIn.tsx [hunk 4]
```diff
@@ -26,12 +26,14 @@ export default function SignIn() {
...
       switchElement.classList.toggle('translate-x-[94%]');
     } else {
-      switchElement.classList.toggle('translate-x-[94%]');
+      console.warn('Switch element not found.');
```
[the remark this answers is not in the document]
Comment (amitojsingh) [reply src/components/signin/SignIn.tsx]: fixed

File: src/pages/dashboard/hooks/useGamePlay1.ts [hunk 5]
```diff
@@ -39,16 +39,21 @@ const useGamePlay = (user: User, toggleLoading: (value: boolean) => void, resetG
...
+          if (
+            gameArray &&
+            gameArray.length > 0 &&
+            currentProgress !== undefined &&
+            currentProgress !== null
+          ) {
             await updateProgress(user.uid, currentProgress, gameArray, currentLevel);
             dispatch(addScreens(gameArray));
           }
           toggleLoading(false);
```
[the remark this answers is not in the document]
Comment (amitojsingh) [reply src/pages/dashboard/hooks/useGamePlay1.ts]: fixed

File: src/pages/profile/index.tsx (hunk 1 above)
Comment (amitojsingh) [reply src/pages/profile/index.tsx]: fixed

**Review by MySmilingTurban — CHANGES_REQUESTED** [review]

**Comment by MySmilingTurban** [comment]
@amitojsingh veerji, do have a final look before merging it

**Pull request merged**